src/Container.js

```diff
@@ -78,6 +91,18 @@ export default class Container {
         };
         this._instanceCache[name] = instance;
     }
+    registerSingleton(name, instance, isExternallyOwned = true) {
```

Comment:
I tried to be as consistent as possible with the other existing methods as `register`.